Add Social Science lessons to lessons page

ErinBecker opened the pull request:
This PR adds a table for the social science lessons to the lessons page. It also moves the Social Science lessons above Geospatial in the list, as they are scheduled for release at an earlier date than the Geospatial lessons.

ErinBecker commented:
@tracykteal @fmichonneau could one of you please look over this and merge?

fmichonneau commented on pages/lessons.md:
```diff
@@ -195,9 +195,62 @@ The course is accessible to:
+<table class="table table-striped">
+    <tr>
+        <th>Lesson</th>
+        <th>Material</th>
```
replace material with "Site" (that's what we use above). Please also change it in the table below.

ErinBecker commented:
Thanks @fmichonneau - I've updated the table as you suggested and also added links to the Reference and Instructor's Guide files where they exist. I'll go ahead and merge this.
